src/Geno/C++/GUI/Widgets/TextEdit.cpp

```diff
@@ -977,6 +997,111 @@ void TextEdit::SetBoxSelection( File& rFile, int LineIndex, float XPosition )
 
 //////////////////////////////////////////////////////////////////////////
 
+void TextEdit::AddCursor( File& rFile, Cursor& rCursor )
+{
+	rCursor.Last = true;
+
+	for( int i = ( int )rFile.Cursors.size() - 1; i >= 0; i-- )
+	{
+		Cursor& rCur = rFile.Cursors[ i ];
+
+		if( rCursor.Position > rCur.Position )
+		{
+			rFile.Cursors.insert( rFile.Cursors.begin() + i + 1, rCursor );
+
+			for( int j = 0; j <= i; j++ )
+			{
+				rFile.Cursors[ j ].Last = false;
+			}
+
+			return;
+		}
+
+		rCur.Last = false;
+	}
+
+	rFile.Cursors.insert( rFile.Cursors.begin(), rCursor );
```

Comment:
Could we use an `std::set` for the cursors so that they are automatically sorted instead?

Reply:
I'm not sure what we would be gaining from that.

Reply:
less code 😛